packages/core/src/actors/promise.ts

```diff
@@ -38,14 +38,64 @@ export type PromiseActorRef<TOutput> = ActorRefFrom<
   PromiseActorLogic<TOutput, unknown>
 >;
 
+/**
+ * An actor logic creator which returns promise logic as defined by an async process that resolves or rejects after some time.
+ *
+ * Actors created from promise actor logic (“promise actors”) can:
+ * - Emit the resolved value of the promise
+ * - Output the resolved value of the promise
+ *
+ * Sending events to promise actors will have no effect.
+ *
+ * @param promiseCreator
+ *   A function which returns a Promise, and accepts an object with the following properties:
+ *   - `input` - Data that was provided to the promise actor
+ *   - `self` - The parent actor of the promise actor
+ *   - `system` - The actor system to which the promise actor belongs
+ * @see {@link https://stately.ai/docs/input | Input docs} for more information about how input is passed
+ *
+ * @example
+ * ```ts
+ * const promiseLogic = fromPromise(() => {
+ *   return fetch('https://example.com/...').then((data) => data.json());
+ * });
```

Comment:
Nit: should we make this an async/await example, as that may be more idiomatic?

```ts
 * const promiseLogic = fromPromise(async () => {
 *   const result = await fetch('https://example.com/...')
 *     .then((data) => data.json());
 * 
 *   return result;
 * });

Reply:
Good call. I'll update.